tests/qml/tst_mainWindowGleanMocks.qml

```diff
@@ -0,0 +1,128 @@
+                spyApplicationId = applicationId
+                spyUploadEnabledInitialize = uploadEnabled
+                spyConfig = config
+            }
+            Glean.initialize = mockGleanInitialize;
+            function mockGleanSetSourceTags(tags) {
+                spyTags = tags
+            }
+            Glean.setSourceTags = mockGleanSetSourceTags;
+            function mockGleanShutdown() {
+                // Should be false before setting it to true in this function.
+                // Helps protect us from bad testing state.
+                compare(spyShutdownCalled, false)
+                spyShutdownCalled = true
+            }
+            Glean.shutdown = mockGleanShutdown
+            function mockSetUploadEnabled(uploadEnabled) {
+                spyUploadEnabled = uploadEnabled
+            }
+            Glean.setUploadEnabled = mockSetUploadEnabled
+        }
+
+        function test_onCompletedCallsMainWindowLoaded() {
+            compare(TestHelper.mainWindowLoadedCalled, true)
+        }
+
+        function test_onSetGleanSourceTagsPassesTagsToGlean() {
+            TestHelper.triggerSetGleanSourceTags("tag1,tag2")
+            compare(spyTags.toString(), "tag1,tag2")
+        }
+
+        function test_onGleanEnabledChangedCallsGleanSetUploadEnabledCorrectly() {
+            VPNSettings.gleanEnabled = false
+            compare(spyUploadEnabled, false)
+
+            VPNSettings.gleanEnabled = true
+            compare(spyUploadEnabled, true)
+        }
+
+        function test_onAboutToQuitCallsGleanShutdown() {
+            TestHelper.triggerAboutToQuit()
+            compare(spyShutdownCalled, true)
+        }
+
+        function test_onInitializeGleanCallsInitializeWithCoreSettings() {
+            TestHelper.triggerInitializeGlean()
+            compare(spyApplicationId, "mozillavpn")
+            compare(spyConfig.appBuild, "MozillaVPN/QMLTest_AppVersion")
+            compare(spyConfig.appDisplayVersion, "QMLTest_AppVersion")
+            // Staging mode is on in the tests
+            compare(spyConfig.channel, "staging")
+            // Debug is on in the tests
+            compare(spyConfig.debug.logPings, true)
+            compare(spyConfig.debug.debugViewTag, "MozillaVPN")
+            // See note in helper.h about this approach for expected values.
+            compare(spyConfig.osVersion, TestHelper.osVersion())
+            compare(spyConfig.architecture, TestHelper.architecture())
+        }
+
+        function test_onInitializeGleanCallsInitializeCorrectUploadEnabled() {
+            VPNSettings.gleanEnabled = false
+            TestHelper.triggerInitializeGlean()
+            compare(spyUploadEnabledInitialize, false)
+
+            VPNSettings.gleanEnabled = true
+            TestHelper.triggerInitializeGlean()
+            compare(spyUploadEnabledInitialize, true)
+        }
+
+        /*
+         * TODO - The following are qml tests that I don't know how to write
```

Comment:
I have added an issue for this TODO https://github.com/mozilla-mobile/mozilla-vpn-client/issues/2296 so it doesn't get lost, and I've added the glean tests to the integration test cases and have a meeting with QA to review the mechanism they need to follow.